src/Fixer/Phpdoc/AlignMultilineCommentFixer.php

```diff
@@ -0,0 +1,165 @@
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class AlignMultilineCommentFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
+{
+    private $tokenKinds;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->tokenKinds = [T_DOC_COMMENT];
+        if ($this->configuration['multiline_comment_blocks']) {
+            $this->tokenKinds[] = T_COMMENT;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Multiline doc comment: enforce asterisk start on each line [PSR-5] and align them.',
+            [
+                new CodeSample(
+'<?php
+    /**
+            * This is a DOC Comment
+with a line not prefixed with asterisk
+
+   */'
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+*/',
+                    ['multiline_comment_blocks' => true]
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+with a line not prefixed with asterisk
+
+   */',
+                    [
+                        'multiline_comment_blocks' => true,
+                        'multiline_comment_blocks_mixed_content' => true,
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_COMMENT, T_DOC_COMMENT]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+        $lineEndingLength = mb_strlen($lineEnding);
+        $indentation = false;
+        foreach ($tokens as $index => $token) {
+            $content = $token->getContent();
+            $newlinePosition = mb_strrpos($content, $lineEnding);
+
+            if (false === $indentation && false !== $newlinePosition) {
+                $indentation = mb_substr($content, $newlinePosition + $lineEndingLength);
+                continue;
+            }
+
+            if (false !== $indentation && isset($content[0]) && trim($content) === '') {
+                $indentation = preg_replace('/[ \t]*[\r\n]+/m', '', $content);
+                continue;
+            }
+
+            if (false !== $indentation && $newlinePosition && $token->isGivenKind($this->tokenKinds)) {
```

Comment:
If I understand correctly, all the code above in the `foreach` tries to find the indentation of the comment, testing each token one by one. Wouldn't it be easier to iterate until you find a comment token and then check the previous one only?